cmd/neofs-lens/internal/object/root.go

```diff
@@ -0,0 +1,278 @@
+package object
+
+import (
+	"fmt"
+	"time"
+
+	common "github.com/nspcc-dev/neofs-node/cmd/neofs-lens/internal"
+	"github.com/nspcc-dev/neofs-node/cmd/neofs-node/config"
+	engineconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine"
+	shardconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine/shard"
+	blobovniczaconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine/shard/blobstor/blobovnicza"
+	fstreeconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine/shard/blobstor/fstree"
+	peapodconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine/shard/blobstor/peapod"
+	"github.com/nspcc-dev/neofs-node/cmd/neofs-node/local_storage"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/blobovniczatree"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/fstree"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/peapod"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine"
+	meta "github.com/nspcc-dev/neofs-node/pkg/local_object_storage/metabase"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/pilorama"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard/mode"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache"
+	"github.com/nspcc-dev/neofs-node/pkg/util"
+	objectSDK "github.com/nspcc-dev/neofs-sdk-go/object"
+	"github.com/panjf2000/ants/v2"
+	"github.com/spf13/cobra"
+	"go.etcd.io/bbolt"
+)
+
+var (
+	vAddress string
+	vOut     string
+	vConfig  string
+)
+var Root = &cobra.Command{
+	Use:   "object",
+	Short: "Operations with an object",
+	Args:  cobra.NoArgs,
+}
+
+func init() {
+	Root.AddCommand(
+		getObjectCMD,
+	)
+}
+
+type shardOptsWithID struct {
+	configID string
+	shOpts   []shard.Option
+}
+type epochState struct {
+}
+
+func (e epochState) CurrentEpoch() uint64 {
+	return 0
+}
+
+func openEngine(cmd *cobra.Command) *engine.StorageEngine {
+	appCfg := config.New(config.Prm{}, config.WithConfigFile(vConfig))
+
+	ls := engine.New()
+
+	var shards []local_storage.ShardCfg
+	err := engineconfig.IterateShards(appCfg, false, func(sc *shardconfig.Config) error {
+		var sh local_storage.ShardCfg
+
+		sh.RefillMetabase = sc.RefillMetabase()
```

Comment:
wow, that does not look like an R/O command that inspects something but more like a real storage engine that may change the underlying storage being inspected. do we really want it? (i did see `ReadOnly` mode but still, it is too dangerous to me, so many routines gonna be started and so many ways the underlying storage may be changed)

cc @roman-khimov, @cthulhu-rider

Reply:
afaik we may have problems with object access with outdated/broken metabase, so this particular settings is worth it

this emulates node's behavior as similar as possible, and i consider such behavior as feature (if Lens have problems then the node will also have)

unsafe? yep, it will always be pretty unsafe, so Lens must work with snapshots only (not current production state)

what we could definitely improve is to make external state basis optional (like GC unconditionally run and based on epochs). IMO this is worth a separate issue

Reply:
I'd expect `lens` to be RO by default. It can have some specific commands that write something, it can have some options to do some changes, but an object getter should be RO by default.

Reply:
i have `shard.WithMode(mode.ReadOnly))`  so I expect everything to be RO.

Reply:
> i have shard.WithMode(mode.ReadOnly)) so I expect everything to be RO.

yes, i saw. mb there is nothing better we can do. but still so many non-RO configs reading copypasta then. also, i have to emphasize that i am afraid when i see that code (so many SE goroutines running and RO flag is the only thing that stops them) reading storage snapshot